server/src/main/java/org/opensearch/index/cache/request/StatsHolder.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.cache.request;
+
+import org.opensearch.common.metrics.CounterMetric;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.common.io.stream.Writeable;
+import org.opensearch.core.common.unit.ByteSizeValue;
+import org.opensearch.core.xcontent.ToXContentFragment;
+import org.opensearch.core.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.io.Serializable;
+
+public class StatsHolder implements Serializable, Writeable, ToXContentFragment {
```

Comment:
Considering this is specific to RequestCacheStats, better to move this inside RequestCacheStats itself.

Reply:
This is also used in ShardRequestCache - I moved it out from this class because I wanted it to be used by RequestCacheStats as well

Reply:
But would this StatsHolder be used elsewhere? As this looks pretty generic though will only be used for requestCache. Considering this is specific to RequestCache, maybe we can still keep it inside ShardRequestCache as public. And use it from RequestCacheStats, should be fine.